src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -760,5 +760,22 @@ internal BoundStatement WrapWithVariablesIfAny(CSharpSyntaxNode scopeDesignator,
                                   ImmutableArray.Create(statement))
                         { WasCompilerGenerated = true };
         }
+
+        /// <summary>
+        /// Should only be used with scopes that could declare local functions.
+        /// </summary>
+        internal BoundStatement WrapWithVariablesAndLocalFunctionsIfAny(CSharpSyntaxNode scopeDesignator, BoundStatement statement)
+        {
+            var locals = this.GetDeclaredLocalsForScope(scopeDesignator);
+            var localFunctions = this.GetDeclaredLocalFunctionsForScope(scopeDesignator);
+            if (locals.IsEmpty && localFunctions.IsEmpty)
+            {
+                return statement;
+            }
+
+            return new BoundBlock(statement.Syntax, locals, localFunctions,
+                                  ImmutableArray.Create(statement))
+            { WasCompilerGenerated = true };
+        }
```

Comment:
The check to know "if the scope could declare local functions" wouldn't be simple. Besides, the code and APIs in Binder are intentionally structured in such a way so that requests for locals and local functions were intentional and only in right contexts.